src/lib.rs

```diff
@@ -223,6 +268,26 @@ fn _keep_app_active(installed_app_id: &str, happs_to_keep: Vec<String>) -> bool
     happs_to_keep.contains(&installed_app_id.to_string()) || installed_app_id.contains("uhCkk")
 }
 
+fn filter_for_hosted_happ_to_uninstall(
+    hosted_app_to_uninstall: Vec<String>,
+    active_apps: Vec<String>,
+) -> Vec<String> {
+    active_apps
+        .into_iter()
+        .filter(|app| {
+            hosted_app_to_uninstall.iter().any(|h| app.starts_with(h))
+                && !app.ends_with("servicelogger")
+        })
+        .collect()
+}
+
+fn filter_for_hosted_happ(active_apps: Vec<String>) -> Vec<String> {
+    active_apps
+        .into_iter()
+        .filter(|app| app.starts_with("uhCkk") && (app.len() == 53))
+        .collect()
+}
```

Comment:
I suggest we extract a function `is_anonymous` and filter by that at the callsite instead.
```rust
fn is_anonymous(&installed_app_id) -> bool {
    app.starts_with("uhCkk") && app.len() == 53
}
````